### PR TITLE
Change optimizer and scheduler configurations

### PR DESCRIPTION
**Pull request by illian01**
## Description

Please include a summary in English, of the changes in this pull request. If it closes an issue, please mention it here.

Closes: #(issue)

You should link at least one existing issue for PR. Before your create a PR, please check to see if there is an issue for this change.  
PRs from forked repository not accepted.

## Change(s)

Optimizer configuration

- Fields such as `opt`, `lr`, `momentum`, and `weight_decay`, which were previously scattered in the training config, are now consolidated under the `optimizer` field.
    - Under the `optimizer` field, a `name` attribute is set to determine which optimizer to use.
    - Parameters required for each optimizer are defined at the same hierarchical level.
- An optimizer builder is modified to create optimizers based on the configuration from `training.optimizer`.
- Configuration update: yaml and cfg module

Learning rate scheduler configuration

- Fields such as `sched`, `min_lr`, `warmup_bias_lr`, etc., which were previously scattered in the training config, are now consolidated under the `scheduler` field.
    - Under the `scheduler` field, a `name` attribute is set to determine which scheduler to use.
    - Parameters required for each scheduler are defined at the same hierarchical level.
- A scheduler builder is modified to create scheduler based on the configuration from `training.scheduler`.
- Configuration update: yaml and cfg module

Fix error missed at #273

- Fix eval resize parameter bug → add missing `max_size`
- Update configs

## Changelog

If you PR to `dev` branch, please add a brief summary of the change to the **Upcoming Release** section of the [`CHANGELOG.md`](https://github.com/Nota-NetsPresso/netspresso-trainer/blob/master/CHANGELOG.md) file and include a link to the PR (formatted in markdown) and a link to your github profile.

For example,

```
- Added a new feature by `@myusername` in [PR 2023](https://github.com/Nota-NetsPresso/netspresso-trainer/pull/2023)
```

## Code Formatting

If you PR to either `master` or `dev` branch, you should follow the code linting process. Please check your code with `lint_check.sh` in `./scripts` directory.
For more information, please read the contribution guide in `CONTRIBUTING.md`.

### CODE REVIEW

File: src/netspresso_trainer/schedulers/builder.py [hunk 1]
```diff
@@ -5,18 +5,14 @@
 
 
 def build_scheduler(optimizer, conf_training):
-    scheduler_name = conf_training.sched
+    conf_scheduler = conf_training.scheduler
+    scheduler_name = conf_scheduler.name
```
Comment (shinkookchoi) [inline src/netspresso_trainer/schedulers/builder.py:9]: I suggest changing `conf_scheduler` to `scheduler_conf` in my opinion.
```suggestion
    scheduler_conf = conf_training.scheduler
    scheduler_name = scheduler_conf.name
```

File: src/netspresso_trainer/schedulers/builder.py [hunk 2]
```diff
@@ -5,18 +5,14 @@
...
-        'total_iters': num_epochs,
-        'iters_per_phase': conf_training.iters_per_phase,  # TODO: config for StepLR
-    })
+
+    # Copy training num_epochs to sub-config scheduler
+    conf_scheduler.total_iters = num_epochs # fix total_iters as num_epochs
 
     assert scheduler_name in SCHEDULER_DICT, f"{scheduler_name} not in scheduler dict!"
-    lr_scheduler = SCHEDULER_DICT[scheduler_name](optimizer, **conf_sched)
+    lr_scheduler = SCHEDULER_DICT[scheduler_name](optimizer, conf_scheduler)
```
Comment (shinkookchoi) [inline src/netspresso_trainer/schedulers/builder.py:16]: I suggest changing `conf_scheduler` to `scheduler_conf` in my opinion, too.

```suggestion
    lr_scheduler = SCHEDULER_DICT[scheduler_name](optimizer, scheduler_conf)
```

**Review by shinkookchoi — COMMENTED** [review]
Overall looks good.
But there are some edits required in a few lines.

**Comment by illian01** [comment]
@shinkookchoi 
Thank you for your detailed reviews. 
I renamed 
- `conf_scheduler` -> `scheduler_conf`
- `conf_training` -> `training_conf`

**Review by shinkookchoi — APPROVED** [review]
Thank you for applying.

**Comment by illian01** [comment]
This pull request contains #264
